src/main/scala-2.10/com/mpc/scalats/core/ScalaParser.scala

```diff
@@ -101,14 +104,24 @@ final class ScalaParser(logger: Logger) {
 
     directKnownSubclasses(tpe) match {
       case possibilities @ (_ :: _ ) => Some(SealedUnion(
-        tpe.typeSymbol.name.toString,
+        buildTypeName(tpe.typeSymbol),
         ListSet.empty ++ members,
         parseTypes(possibilities)))
 
       case _ => Option.empty[SealedUnion]
     }
   }
 
+  private def parseEnumeration(enumerationValueType: Type): Option[Enumeration] = {
```

Comment:
that should be only in the other PR

Reply:
Done, now this PR is based directly on master